src/views/ProductDetail.vue

```diff
@@ -97,43 +69,151 @@
 import {
   IonBackButton,
   IonButton,
-  IonCard,
   IonChip,
   IonContent,
   IonHeader,
   IonItem,
-  IonItemDivider,
   IonLabel,
   IonList,
   IonListHeader,
   IonNote,
   IonPage,
+  IonRow,
   IonTitle,
-  IonToolbar
+  IonToolbar,
+  modalController
 } from "@ionic/vue";
 import { defineComponent } from "vue";
+import { mapGetters, useStore } from "vuex";
 import Image from "../components/Image.vue";
+import { StockService } from '@/services/StockService'
+import { hasError, getFeature, showToast } from "@/utils";
+import { sortSizes } from '@/apparel-sorter';
+import OtherStoresInventoryModal from "./OtherStoresInventoryModal.vue";
+import { translate } from "@/i18n";
 
 export default defineComponent({
   name: "ProductDetail",
   components: {
     IonBackButton,
     IonButton,
-    IonCard,
     IonChip,
     IonContent,
     IonHeader,
     IonItem,
-    IonItemDivider,
     IonLabel,
     IonList,
     IonListHeader,
     IonNote,
     IonPage,
+    IonRow,
     IonTitle,
     IonToolbar,
     Image
   },
+  data() {
+    return {
+      selectedColor: '',
+      selectedSize: '',
+      features: [] as any,
+      currentVariant: {} as any,
+      currentStoreInventory: 0,
+      otherStoresInventory: 0,
+      warehouseInventory: 0,
+      otherStoresInventoryDetails: [] as any
+    }
+  },
+  computed: {
+    ...mapGetters({
+      product: "product/getCurrent",
+      currentFacility: 'user/getCurrentFacility',
+    })
+  },
+  async beforeMount() {
+    await this.store.dispatch('product/getVariants', { productId: this.$route.params.productId })
+    this.getFeatures()
```

Comment:
I think we should handle case for the product with no selctectable features

Reply:
Updated sir